java/src/main/java/com/codurance/training/tasks/TaskList.java

```diff
@@ -15,8 +19,11 @@ public final class TaskList implements Runnable {
     private final Map<String, List<Task>> tasks = new LinkedHashMap<>();
     private final BufferedReader in;
     private final PrintWriter out;
-
-    private long lastId = 0;
+    private AddServiceImpl addService;
+    private CheckServiceImpl checkService;
+    private DeleteServiceImpl deleteService;
+    private ViewServiceImpl viewService;
+    private TodayDueTaskServiceImpl todayDueTaskService;
```

Comment:
make them final

Reply:
use AddService instead of AddServiceImpl
Similarly for others
so class depends on abstraction than on concrete class

Reply:
Made them final and used AddService instead of AddServiceImpl